### PR TITLE
Release fixes for asynchronous crypto

### PR DESCRIPTION
**Pull request by dgarske**
* Fix for async TLS v1.3 post handshake auth. The re-handshake was not resetting the `processReply` state.
* Fix asyncDev usage in test.c
* Fix in `test_wolfSSL_OBJ_ln()` in api.c for handling `ECC_CURVE_INVALID`.
* Fixes for Cavium Nitrox and Intel QuickAssist.
* Fix api.c tests using "free()" instead of "XFREE" causing issues with custom allocators.

### CODE REVIEW

File: src/tls13.c [hunk 1]
```diff
@@ -7180,40 +7180,42 @@ int DoTls13HandShakeMsgType(WOLFSSL* ssl, byte* input, word32* inOutIdx,
...
-            if (type == server_hello) {
-                if ((ret = DeriveEarlySecret(ssl)) != 0)
-                    return ret;
-                if ((ret = DeriveHandshakeSecret(ssl)) != 0)
-                    return ret;
-
-                if ((ret = DeriveTls13Keys(ssl, handshake_key,
-                                           ENCRYPT_AND_DECRYPT_SIDE, 1)) != 0) {
-                    return ret;
+            if (ssl->options.handShakeState != CLIENT_HELLO_COMPLETE) {
```
Comment (SparkiDev) [inline src/tls13.c:7183]: Why is this necessary?
The ret is checked for 0 and the message type is server_hello or finished.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: src/tls13.c (hunk 1 above)
Comment (dgarske) [reply src/tls13.c:7183]: I made this change to prevent having to re-calculate this on each re-entry during the re-handshake. Let me test without it and see if it's really being called multiple. Now that you point me to it I think the `type` will do that properly.

**Review by SparkiDev — APPROVED** [review]
TLS 1.3 part is good.

**Review by toddouska — APPROVED** [review]
